### ISSUE TITLE
Flaky errors when executed in GitHub Actions

### ISSUE THREAD

**Issue by ptrdom** [VirtusLab/scala-cli#932]
**Version(s)**

`0.1.2` - https://github.com/VirtusLab/scala-cli-setup GitHub Action is used for installation.
`0.1.4` - manual installation.

**Describe the bug**

The execution of `scala-cli test` seems flaky in GitHub Actions - sometimes it succeeds and sometimes fails by throwing one of two different errors:
```
0DException in thread "main" java.util.concurrent.TimeoutException: Futures timed out after [30 seconds]
	at scala.concurrent.impl.Promise$DefaultPromise.ready(Promise.scala:259)
	at scala.concurrent.impl.Promise$DefaultPromise.result(Promise.scala:263)
	at scala.concurrent.Await$.$anonfun$result$1(package.scala:223)
	at scala.concurrent.BlockContext$DefaultBlockContext$.blockOn(BlockContext.scala:57)
	at scala.concurrent.Await$.result(package.scala:146)
	at scala.build.blooprifle.internal.Operations$.timeout(Operations.scala:373)
	at scala.build.blooprifle.internal.Operations$.about(Operations.scala:341)
	at scala.build.blooprifle.BloopRifle$.getCurrentBloopVersion(BloopRifle.scala:192)
	at scala.build.bloop.BloopServer$.ensureBloopRunning(BloopServer.scala:112)
	at scala.build.bloop.BloopServer$.bsp(BloopServer.scala:153)
	at scala.build.bloop.BloopServer$.buildServer(BloopServer.scala:183)
	at scala.build.bloop.BloopServer$.withBuildServer(BloopServer.scala:248)
	at scala.build.Build$.build(Build.scala:422)
	at scala.build.Build$.build(Build.scala:449)
	at scala.cli.commands.Test$.run(Test.scala:120)
	at scala.cli.commands.Test$.run(Test.scala:16)
	at caseapp.core.app.CaseApp.main(CaseApp.scala:149)
	at caseapp.core.app.CommandsEntryPoint.main(CommandsEntryPoint.scala:116)
	at scala.cli.ScalaCliCommands.main(ScalaCliCommands.scala:89)
	at scala.cli.ScalaCli$.main0(ScalaCli.scala:148)
	at scala.cli.ScalaCli$.main(ScalaCli.scala:71)
	at scala.cli.ScalaCli.main(ScalaCli.scala)
```
```
0DException in thread "main" java.lang.RuntimeException: Fatal error, could not spawn Bloop: ParsingFailed()
	at scala.build.bloop.BloopServer$.$anonfun$ensureBloopRunning$8(BloopServer.scala:113)
	at scala.util.Either.fold(Either.scala:192)
	at scala.build.bloop.BloopServer$.ensureBloopRunning(BloopServer.scala:113)
	at scala.build.bloop.BloopServer$.bsp(BloopServer.scala:153)
	at scala.build.bloop.BloopServer$.buildServer(BloopServer.scala:183)
	at scala.build.bloop.BloopServer$.withBuildServer(BloopServer.scala:248)
	at scala.build.Build$.build(Build.scala:422)
	at scala.build.Build$.build(Build.scala:449)
	at scala.cli.commands.Test$.run(Test.scala:120)
	at scala.cli.commands.Test$.run(Test.scala:16)
	at caseapp.core.app.CaseApp.main(CaseApp.scala:149)
	at caseapp.core.app.CommandsEntryPoint.main(CommandsEntryPoint.scala:116)
	at scala.cli.ScalaCliCommands.main(ScalaCliCommands.scala:89)
	at scala.cli.ScalaCli$.main0(ScalaCli.scala:148)
	at scala.cli.ScalaCli$.main(ScalaCli.scala:71)
	at scala.cli.ScalaCli.main(ScalaCli.scala)
```

**To Reproduce**

You can see this bug being reproduced in the GitHub Actions of this PR - https://github.com/playframework/play-scala-react-seed/pull/86.

**Comment by ptrdom** [comment]
Might be Bloop's issue, posted a question in gitter - https://gitter.im/scalacenter/bloop?at=6265572bc43500250055a629.

**Comment by alexarchambault** [comment]
Thanks for reporting this @ptrdom.

Would it be possible to run the `scala-cli` command with `-v -v -v -v` (4 `-v`)? The output of a failure with that verbosity level could help us debugging that.

**Comment by lwronski** [comment]
Hi @ptrdom, thanks also for reporting.

I updated ScalaCLI version in `scala-cli-setup` to [0.1.4](https://github.com/VirtusLab/scala-cli-setup/commit/78483860002f6114ed87aca41779f54dd7370377). Now `scala-cli-setup` should use the newest version of `scala-cli`.

**Comment by alexarchambault** [comment]
@ptrdom  In the mean time, running the Scala CLI with `--server=false` could help in your case (this makes Scala CLI run scalac itself, rather than relying on Bloop), but wouldn't help us understand the problem with Bloop here.

**Comment by alexarchambault** [comment]
Also, Scala CLI uses [its own fork of Bloop](https://github.com/scala-cli/bloop-core) for now, that it connects to differently than how Bloop mainline does, so I don't know if people there can help (and note that the Bloop discussion happens more on the official Scala Discord these days).

**Comment by ptrdom** [comment]
Thank you for replies!
I will try reproducing the issue with `-v -v -v -v` and post back the results, then use `--server=false` until the issue gets resolved.
It would probably make sense to have a `scala-cli` version parameter exposed to https://github.com/VirtusLab/scala-cli-setup  action's users, right?
Also, I have remembered after posting in gitter that Scala's core community moved to Discord recently, so I posted about the issue there too - https://discord.com/channels/632150470000902164/884881696300859442/967788620842946620. Nice to see a channel for `scala-cli` there 👍
